pkg/raft/raft.go

```diff
@@ -637,6 +637,19 @@ func (r *raft) makeMsgApp(to pb.PeerID, pr *tracker.Progress, ls logSlice) pb.Me
 	}
 }
 
+// maybeMakeMsgApp returns a MsgApp message to be sent to the given peer, or a
+// zero-value message if it can not be sent.
```

Comment:
"if it can not be sent" is misleading or at least a bit unhelpful - would it be better to say ", or a zero-value message if instance's state or tracking status does not permit this"?

Also, would it be better to return `(pb.Message, bool)`? A caller to `maybeMakeMsgApp` might expect to get one either way, and not an empty message. If they send it anyway, that could cause issues.